core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -474,6 +475,12 @@ final class ShuffleBlockFetcherIterator(
               // TODO: manage the memory used here, and spill it into disk in case of OOM.
               Utils.copyStream(input, out, closeStreams = true)
               input = out.toChunkedByteBuffer.toInputStream(dispose = true)
+            } else {
+              val readAheadEnabled = SparkEnv.get != null &&
+                SparkEnv.get.conf.get(config.SHUFFLE_READ_AHEAD)
+              if (readAheadEnabled) {
+                input = new ReadAheadInputStream(input, 1024 * 1024)
```

Comment:
I am just wondering why and how 1MB is chosen here. I know in Spark we have already too many config values but what is your opinion using only one config  `spark.shuffle.read.ahead.buffer.size` which could be `bytesConf` and when it is 0 then wrapping into `ReadAheadInputStream` is skipped (like in case of disabled now).

Reply:
That sounds good，thanks

Reply:
IIUC this won't be enabled when `detectCorruptUseExtraMemory` is set.
Would it make sense to document this?

Reply:
yes I think we should document this behavior, also agree with some of the other comments not sure we need the enabled config, perhaps the buffer size one and if its set to 0 then disabled.  How did you choose the buffer size, just take it from the existing spill reader config? did you try different buffer sizes?

Reply:
Thanks all, I think  @attilapiros 's advice is better.
I've tried 1M, 2M, 4M sizes, however, no significant performance differences were observed.